tests/zkevm/test_worst_compute.py

```diff
@@ -234,23 +235,167 @@ def test_worst_modexp(
+                "103188585E2364128FE25C70558F1560F4F9350BAF3959E603CC91486E110936",
+                "198E9393920D483A7260BFB731FB5D25F1AA493335A9E71297E485B7AEF312C2",
+                "1800DEEF121F1E76426A00665E5C4479674322D4F75EDADD46DEBD5CD992F6ED",
+                "090689D0585FF075EC9E99AD690C3395BC4B313370B38EF355ACDADCD122975B",
+                "12C85EA5DB8C6DEB4AAB71808DCB408FE3D1E7690C43D37B4CE6CC0166FA7DAA",
+            ],
+            id="bn128_pairing",
+        ),
+        pytest.param(
+            0x09,
+            [
+                "0000FFFF",
+                "48C9BDF267E6096A3BA7CA8485AE67BB2BF894FE72F36E3CF1361D5F3AF54FA5D182E6AD7F520E511F6C3E2B8C68059B6BBD41FBABD9831F79217E1319CDE05B",
+                "6162630000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+                "00000000000300000000000000000000",
+                "01",
+            ],
+            id="blake2f",
+        ),
+        pytest.param(
+            0x0A,
+            [
+                "01E798154708FE7789429634053CBF9F99B619F9F084048927333FCE637F549B",
+                "564C0A11A0F704F4FC3E8ACFE0F8245F0AD1347B378FBF96E206DA11A5D36306",
+                "24D25032E67A7E6A4910DF5834B8FE70E6BCFEEAC0352434196BDF4B2485D5A1",
+                "8F59A8D2A1A625A17F3FEA0FE5EB8C896DB3764F3185481BC22F91B4AAFFCCA25F26936857BC3A7C2539EA8EC3A952B7",
+                "873033E038326E87ED3E1276FD140253FA08E9FC25FB2D9A98527FC22A2C9612FBEAFDAD446CBC7BCDBDCD780AF2C16A",
+            ],
+            id="point_evaluation",
+        ),
+        pytest.param(
+            bls12381_spec.Spec.G1ADD,
+            [
+                bls12381_spec.Spec.G1,
+                bls12381_spec.Spec.P1,
+            ],
+            id="bls12_g1add",
+        ),
+        pytest.param(
+            bls12381_spec.Spec.G1MSM,
+            [
+                (bls12381_spec.Spec.P1 + bls12381_spec.Scalar(bls12381_spec.Spec.Q))
+                * (len(bls12381_spec.Spec.G1MSM_DISCOUNT_TABLE) - 1),
+            ],
+            id="bls12_g1msm",
+        ),
+        pytest.param(
+            bls12381_spec.Spec.G2ADD,
+            [
+                bls12381_spec.Spec.G2,
+                bls12381_spec.Spec.P2,
+            ],
+            id="bls12_g2add",
+        ),
+        pytest.param(
+            bls12381_spec.Spec.G2MSM,
+            [
+                # TODO: the //2 is required due to a limitation of the max contract size limit.
+                # In a further iteration we can insert the inputs as calldata or storage and avoid
+                # having to do PUSHes which has this limtiation. This also applies to G1MSM.
+                (bls12381_spec.Spec.P2 + bls12381_spec.Scalar(bls12381_spec.Spec.Q))
+                * (len(bls12381_spec.Spec.G2MSM_DISCOUNT_TABLE) // 2),
```

Comment:
We can improve on this in a further PR. With this style the RISCV emulator already crashes.